fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/RuleSet.java

```diff
@@ -199,10 +207,12 @@
                 new PushDownPredicateIntersectRule(),
                 new PushDownPredicateTableFunctionRule(),
                 new PushDownPredicateRepeatRule(),
+
                 MergePredicateRule.HIVE_SCAN,
                 MergePredicateRule.SCHEMA_SCAN,
                 MergePredicateRule.MYSQL_SCAN,
-                new MergeTwoFiltersRule()
+                new MergeTwoFiltersRule(),
+                new PushDownPredicateCTEConsumeRule()
```

Comment:
why some rules use static instances and another use new to constructe objects?

Reply:
Some static instances just for reuse code. In fact, many rule-list is singletons in `RuleSet`, so these ways are the same. 

Of course, we want to organize all the rules into singletons later.

Reply:
It seems reasonable